auth0_flutter/example/lib/example_app.dart

```diff
@@ -21,29 +23,40 @@ class _ExampleAppState extends State<ExampleApp> {
 
   late Auth0 auth0;
   late WebAuthentication webAuth;
+  late Auth0Web auth0Web;
 
   @override
   void initState() {
     super.initState();
     auth0 = Auth0(dotenv.env['AUTH0_DOMAIN']!, dotenv.env['AUTH0_CLIENT_ID']!);
+
+    auth0Web =
+        Auth0Web(dotenv.env['AUTH0_DOMAIN']!, dotenv.env['AUTH0_CLIENT_ID']!);
+
     webAuth =
         auth0.webAuthentication(scheme: dotenv.env['AUTH0_CUSTOM_SCHEME']);
+
+    auth0Web.onLoad();
   }
 
   Future<void> webAuthLogin() async {
-    String output;
-
-    // Platform messages may fail, so we use a try/catch PlatformException.
-    // We also handle the message potentially returning null.
-    try {
-      final result = await webAuth.login();
-      output = result.idToken;
-
-      setState(() {
-        _isLoggedIn = true;
-      });
-    } on WebAuthenticationException catch (e) {
-      output = e.toString();
+    String output = '';
+
+    if (kIsWeb) {
+      auth0Web.loginWithRedirect(redirectUri: 'http://localhost:3000');
```

Comment:
Should this have an `await`, and be all wrapped in the same try/catch?

Reply:
Also, what happens with the errors returned from JS land?

Reply:
> Should this have an await, and be all wrapped in the same try/catch?

Yes, probably should.

> Also, what happens with the errors returned from JS land?

They _should_ get surfaced in the same way. Something to test, though.